src/main/java/decodes/cwms/CwmsTimeSeriesDb.java

```diff
@@ -859,6 +859,9 @@ public DbKey connect( String appName, Properties credentials )
 				+ " with officeID=" + dbOfficeId);
 
 				postConnectInit(appName, conn); // Make sure the versions and such are set
+				setConnection(conn);
+				setupKeyGenerator(); // TODO: this should take the connection as a parameter
```

Comment:
looks easy enough to overload and deprecate

Reply:
overloads... why do I always forget overloads.

Reply:
actually I take that back, `KeyGeneratorFactory.makeKeyGenerator` isn't even using the parameterized connection which means the `setConnection(conn)` and `setConnection(null)` aren't needed

Reply:
Still this last comment

Reply:
Yeah, I'll look into that more, there's some downstream stuff that might still use it so I was a little reluctant to change it more but a simple find reference should help with that.

Reply:
Oh wait, I'll still look, but if you look at say, the "OpenTSDBDB" implementation it still expects the connection to be set once, but I could probably get away with just not calling those in this implementation.

Reply:
Nevermind, that was all fairly easy to change, HDB does it's own thing so setKeyGenerator is only used in two places.